src/index.ts

```diff
@@ -20,14 +20,71 @@ import {
   RepoDomain,
   BranchDomain,
   FileData,
+  RawContent,
+  CreateReviewCommentUserOptions,
 } from './types';
 export {Changes} from './types';
 import {Octokit} from '@octokit/rest';
 import {Logger, LoggerOptions} from 'pino';
 import {logger, setupLogger} from './logger';
-import {addPullRequestDefaults} from './default-options-handler';
+import {
+  addPullRequestDefaults,
+  addReviewCommentsDefaults,
+} from './default-options-handler';
 import * as retry from 'async-retry';
 
+/**
+ * Given a set of suggestions, make all the multiline inline review comments on a given pull request given
+ * that they are in scope of the pull request. Outof scope suggestions are not made.
+ *
+ * In-scope suggestions are specifically: the suggestion for a file must correspond to a file in the remote pull request
+ * and the diff hunk computed for a raw file must produce a range that is a subset of the pull request's files hunks.
+ *
+ * If a file is too large to load in the review, it is skipped in the suggestion phase.
+ *
+ * If changes are empty then the workflow will not run.
+ * Rethrows an HttpError if Octokit GitHub API returns an error. HttpError Octokit access_token and client_secret headers redact all sensitive information.
+ * @param octokit The authenticated octokit instance, instantiated with an access token having permissiong to create a fork on the target repository.
+ * @param rawChanges A set of changes. The changes may be empty.
+ * @param options The configuration for interacting with GitHub provided by the user.
+ * @param loggerOption The logger instance (optional).
+ */
+export async function reviewPullRequest(
+  octokit: Octokit,
+  rawChanges: Map<string, RawContent>,
+  options: CreateReviewCommentUserOptions,
+  loggerOption?: Logger | LoggerOptions
+): Promise<void> {
```

Comment:
Let's return the pull request number (or maybe the review id?) on success like we do for `createPullRequest()`. It can signal whether or not we took an action on the pull request.